### PR TITLE
Syncs the HTMLAllCollection to latest WHATWG status

### LINKED ISSUE #120: Make changes to HTMLAllCollection to track recent discussion about making the WebIDL match reality
See Issue: https://github.com/whatwg/html/commit/4315a98056c60637546f739c2870b6945136af6d

### PR DESCRIPTION
**Pull request by travisleithead**
Fixes #120

### CODE REVIEW

File: sections/infrastructure.include [hunk 1]
```diff
@@ -3836,95 +3834,91 @@
...
+    elements is returned.
 
-      Only <{button}>, <{form}>, <{iframe}>, <{input}>,
-      <{map}>, <{meta}>, <{object}>, <{select}>, and
-      <{textarea}> elements can have a name for the purpose of this method; their name is
-      given by the value of their <code>name</code> attribute.
-    </dd>
+    Only <{button}>, <{form}>, <{iframe}>, <{input}>, <{map}>, <{meta}>, <{object}>, <{select}>, and
+    <{textarea}> elements can have a name for the purpose of this method; their name is given by the
+    value of their <code>name</code> attribute.</dd>
```
Comment (arronei) [inline sections/infrastructure.include]: You can link the name attribute too

Comment (travisleithead) [inline sections/infrastructure.include]: Yuck. Linking name in this context proved challenging.

**Comment by travisleithead** [comment]
Addressed comment.

File: sections/infrastructure.include [hunk 2]
```diff
@@ -3836,95 +3834,93 @@
...
-  <{a}>, <{applet}>, <{button}>, <{embed}>, <{form}>, <{frame}>, <{frameset}>, <{iframe}>,
-  <{img}>, <{input}>, <{map}>, <{meta}>, <{object}>, <{select}>, and <{textarea}>
-
   The <a spec="webidl">supported property names</a> consist of the non-empty values of all the
-  <code>id</code> attributes of all the elements <a>represented by the collection</a>, and the
-  non-empty values of all the <code>name</code> attributes of all the <a>"all"-named element or
-  elements</a> <a>represented by the collection</a>, in <a>tree order</a>, ignoring later
-  duplicates, with the <code>id</code> of an element preceding its <code>name</code> if it contributes both, they
-  differ from each other, and neither is the duplicate of an earlier entry.
+  <{global/id}> and <code>name</code> attributes of all the elements
```
Comment (arronei) [inline sections/infrastructure.include]: Create a generic name for the group of name attribute elements.

Comment (travisleithead) [inline sections/infrastructure.include]: OK.

Note: this is more widespread in this section (all collections of various flavors use <code>name</code>)

**Comment by arronei** [comment]
:+1:

**Pull request merged**